python/tests/test_parameter.py

```diff
@@ -23,3 +23,15 @@ def test_linear(state):
 
     linear = flyft.parameter.LinearParameter(3.0, 1.0, 4.5)
     assert linear(state) == pytest.approx(3.0+4.5*(2.0-1.0))
+
+def test_squareroot(state):
+    state.time = 2.0
+
+    squareroot = flyft.parameter.SquareRootParameter(3.0, 1.0, 4.5)
+    assert squareroot(state) == pytest.approx(np.sqrt(3.0*3.0+4.5*(2.0-1.0)))
+
+def test_squareroot_exception(state):
+    with pytest.raises(ValueError):
+        state.time = -1.0
+        squareroot = flyft.parameter.SquareRootParameter(3.0, 1.0, 4.5)
+        squareroot(state)
```

Comment:
You only expect a value error when you actually evaluate, so do like this:
```suggestion
    state.time = -1.0
    squareroot = flyft.parameter.SquareRootParameter(3.0, 1.0, 4.5)
    with pytest.raises(ValueError):
        squareroot(state)
```
Note also you should have a test where t is a valid (>= 0) time, but the square root cannot evaluate because the argument to the square root goes negative.